spec/integration/active_storage/rails_app.rb

```diff
@@ -0,0 +1,112 @@
+require 'bundler/inline'
+
+gemfile(true) do
+  source 'https://rubygems.org'
+
+  git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+  gem 'rails', '6.0.3'
+  gem 'sqlite3'
+  gem 'aws-sdk-s3', require: false
+  gem 'format_parser', path: './'
+end
+
+require 'active_record/railtie'
+require 'active_storage/engine'
+require 'tmpdir'
+
+class TestApp < Rails::Application
+  config.root = __dir__
+  config.hosts << 'example.org'
+  config.eager_load = false
+  config.session_store :cookie_store, key: 'cookie_store_key'
+  secrets.secret_key_base = 'secret_key_base'
+
+  config.logger = Logger.new('/dev/null')
+
+  if ENV['S3']
+    # We simulate local S3 using Minio
+    config.active_storage.service = :minio
+    config.active_storage.service_configurations = {
+      minio: {
+        service: 'S3',
+        access_key_id: 'weak_key',
+        secret_access_key: 'weak_key',
+        region: 'eu-west-1',
+        bucket: 'format-parser-testing-tmp',
+        endpoint: 'http://127.0.0.1:9000',
+        force_path_style: true
+      }
+    }
+  else
+    config.active_storage.service = :local
+    config.active_storage.service_configurations = {
+      local: {
+        root: Dir.tmpdir,
+        service: 'Disk'
+      }
+    }
+  end
+
+  config.active_storage.analyzers.prepend FormatParser::ActiveStorage::BlobAnalyzer
+end
+
+ENV['DATABASE_URL'] = 'sqlite3::memory:'
+
+Rails.application.initialize!
+
+require ActiveStorage::Engine.root.join('db/migrate/20170806125915_create_active_storage_tables.rb').to_s
+
+ActiveRecord::Schema.define do
+  CreateActiveStorageTables.new.change
+
+  create_table :users, force: true
+end
+
+class User < ActiveRecord::Base
+  has_one_attached :profile_picture
+end
+
+require 'minitest/autorun'
+require 'open-uri'
+
+describe User do
+  describe "profile_picture's metadatas" do
+    before do
+      next unless ENV['S3']
+
+      unless `ps aux | grep '[m]inio server'`.present?
```

Comment:
I am not super fan of those extra step here but I didn't wanted to pollute `spec` folder or to put this code globally.